airflow/providers/amazon/aws/operators/glue_session.py

```diff
@@ -0,0 +1,260 @@
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.glue_session import (
+    GlueSessionHook,
+    GlueSessionProtocol,
+    GlueSessionStates,
+)
+from airflow.providers.amazon.aws.operators.base_aws import AwsBaseOperator
+from airflow.providers.amazon.aws.triggers.glue_session import GlueSessionReadyTrigger
+from airflow.utils.helpers import prune_dict
+
+if TYPE_CHECKING:
+    import boto3
+
+    from airflow.utils.context import Context
+
+
+class GlueSessionBaseOperator(AwsBaseOperator[GlueSessionHook]):
+    """This is the base operator for all Glue service operators."""
+
+    aws_hook_class = GlueSessionHook
+
+    @cached_property
+    def client(self) -> GlueSessionProtocol | boto3.client:
+        """Create and return the GlueSessionHook's client."""
+        return self.hook.conn
+
+    def _complete_exec_with_session(self, context, event=None):
+        """To be used as trigger callback for operators that return the session description."""
+        if event["status"] != "success":
+            raise AirflowException(f"Error while waiting for operation on session to complete: {event}")
+        session_id = event.get("id")
+        # We cannot get the cluster definition from the waiter on success, so we have to query it here.
+        details = self.hook.conn.get_session(Id=session_id)["Session"]
+        return details
+
+
+class GlueCreateSessionOperator(GlueSessionBaseOperator):
+    """Create an AWS Glue Session.
+
+    AWS Glue is a serverless Spark ETL service for running Spark Jobs on the AWS
+    cloud. Language support: Python and Scala.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GlueCreateSessionOperator`
+
+    :param session_id: session id
+    :param session_desc: session description
+    :param region_name: aws region name (example: us-east-1)
+    :param iam_role_name: AWS IAM Role for Glue Session Execution. If set `iam_role_arn` must equal None.
+    :param iam_role_arn: AWS IAM Role ARN for Glue Session Execution, If set `iam_role_name` must equal None.
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Session
+    :param create_session_kwargs: Extra arguments for Glue Session Creation
+    """
+
+    template_fields: Sequence[str] = (
+        "session_id",
+        "create_session_kwargs",
+        "iam_role_name",
+        "iam_role_arn",
+    )
+    template_fields_renderers = {
+        "create_session_kwargs": "json",
+    }
+
+    def __init__(
+        self,
+        *,
+        session_id: str = "aws_glue_default_session",
+        session_desc: str = "AWS Glue Session with Airflow",
+        aws_conn_id: str = "aws_default",
+        iam_role_name: str | None = None,
+        iam_role_arn: str | None = None,
+        num_of_dpus: int | None = None,
+        create_session_kwargs: dict | None = None,
+        wait_for_readiness: bool = True,
```

Comment:
We usually call this wait_for_completion

Also can you document this in the doc string?